File: 0782-jewels-and-stones/0782-jewels-and-stones.cpp
#include<unordered_map>

class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_map<char, int> jewel;
        unordered_map<char, int> stone;
        int ret = 0;

        for (char val : jewels) {
            jewel[val]++;
        }
        for (char val : stones) {
            stone[val]++;
        }

       

        for (auto val : stone) {
            if (jewel.find(val.first) != jewel.end()) {
                ret += val.second;
            }
        }

        return ret;
    }
};

File: 0782-jewels-and-stones/0782-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        bag: dict[str, int] = {}
        ret: int = 0
        for jewel in jewels:
            bag[jewel] = bag.get(jewel,0)+1
        for stone in stones:
            ret += bag.get(stone,0)
        return ret